.github/workflows/build.yml

```diff
@@ -45,3 +45,18 @@ jobs:
         run: xcodebuild -project sample/app-ios/app-ios.xcodeproj -scheme app-ios -sdk iphonesimulator -arch x86_64 build
       - name: Build iOS Compose sample
         run: xcodebuild -project sample/app-ios-compose/app-ios-compose.xcodeproj -scheme app-ios-compose -sdk iphonesimulator -arch x86_64 build
+  check-binary-compatibility:
+    name: Check binary compatibility
+    runs-on: macos-14
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: 17
+      - name: Check binary compatibility
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: apiCheck
```

Comment:
I think this should be already running as part of the `build` task. We may need additional changes in the `gradle-setup-plugin` to only run `apiCheck` on macOS hosts. Currently, it is [running only on Linux](https://github.com/arkivanov/gradle-setup-plugin/blob/b42078d4592d0a59c24d86805202d0e66f2efdb1/src/main/java/com/arkivanov/gradle/SetupBinaryCompatibilityValidator.kt#L9) hosts on CI builds.